// usebootstrap/dist/runtime/components/bootstrap/dropdown/Dropdown.js
import { hProps, exposeMethods, addProp } from "../../../composables/utils/useProps.js";
import { useBlock, BlockProps } from "../../../composables/base/useBlock.js";
import { DropdownProps, useDropdown } from "../../../composables/bootstrap/useDropdown.js";
import Fragment from "../../vue/Fragment.vue";
import { defineComponent, h, ref, computed, inject } from "#imports";
export default defineComponent({
  name: "Dropdown",
  props: {
    ...BlockProps,
    ...DropdownProps,
    tag: {
      type: String,
      default: "div"
    },
    borderRadius: {
      type: String,
      default: void 0
    }
  },
  setup(props, context) {
    const elementRef = ref();
    const block = useBlock(props);
    const dropdown = useDropdown(props, context, elementRef);
    const isInputGroup = inject("isInputGroup", false);
    exposeMethods(context, dropdown, block);
    const current = {
      class: {
        dropdown: !isInputGroup
      },
      style: computed(() => {
        return {
          ...addProp(props.borderRadius, "--bs-dropdown-border-radius", `${props.borderRadius}`)
        };
      }),
      ref: elementRef
    };
    return () => h(
      isInputGroup ? Fragment : props.tag,
      hProps(block, dropdown, current),
      // current は最後に
      context.slots
    );
  }
});
