// usebootstrap/dist/runtime/components/bootstrap/dropdown/DropdownMenu.js
import { hProps, addProp } from "../../../composables/utils/useProps.js";
import { useBlock, BlockProps } from "../../../composables/base/useBlock.js";
import {
  DropdownMenuProps,
  useDropdownMenu
} from "../../../composables/bootstrap/useDropdownMenu.js";
import { defineComponent, h, ref, computed } from "#imports";
export default defineComponent({
  name: "DropdownMenu",
  props: {
    ...BlockProps,
    ...DropdownMenuProps,
    tag: {
      type: String,
      default: "ul"
    },
    borderRadius: {
      type: String,
      default: void 0
    }
  },
  setup(props, context) {
    const elementRef = ref();
    const block = useBlock(props);
    const dropdownMenu = useDropdownMenu(props, elementRef);
    const current = {
      style: computed(() => {
        return {
          ...addProp(props.borderRadius, "--bs-dropdown-border-radius", `${props.borderRadius}`)
        };
      }),
      ref: elementRef
    };
    return () => {
      return h(props.tag, hProps(current, block, dropdownMenu), context.slots);
    };
  }
});
