// usebootstrap/dist/runtime/modules/useComponents.js
import { pickBy } from "lodash-es";
export const selectComponents = (category) => {
  return pickBy(components, (component) => component.category && component.category === category);
};
export const getComponent = (name) => {
  return components[name];
};
export const components = {
  Accordion: {
    path: "./runtime/components/bootstrap/accordion/Accordion",
    category: "bootstrap",
    examples: {
      test: {
        items: [
          {
            component: "AccordionSection",
            text: "AccordionSection",
            attrs: {
              title: "AccordionSection #1"
            }
          }
        ]
      }
    },
    props: {
      parent: {
        type: Boolean,
        default: false
      },
      color: {
        type: String,
        default: void 0
      },
      activeColor: {
        type: String,
        default: void 0
      },
      activeBackgroundColor: {
        type: String,
        default: void 0
      },
      activeButtonColor: {
        type: String,
        default: void 0
      },
      buttonIconWidth: {
        type: String,
        default: void 0
      },
      backgroundColor: {
        type: String,
        default: void 0
      }
    }
  },
  AccordionBody: {
    path: "./runtime/components/bootstrap/accordion/AccordionBody",
    category: "bootstrap",
    spec: {
      parents: ["Accordion"]
    }
  },
  AccordionButton: {
    path: "./runtime/components/bootstrap/accordion/AccordionButton",
    category: "bootstrap",
    spec: {
      parents: ["Accordion"]
    }
  },
  AccordionCollapse: {
    path: "./runtime/components/bootstrap/accordion/AccordionCollapse",
    category: "bootstrap",
    spec: {
      parents: ["Accordion"]
    }
  },
  AccordionHeader: {
    path: "./runtime/components/bootstrap/accordion/AccordionHeader",
    category: "bootstrap",
    spec: {
      parents: ["Accordion"]
    }
  },
  AccordionItem: {
    path: "./runtime/components/bootstrap/accordion/AccordionItem",
    category: "bootstrap",
    spec: {
      parents: ["Accordion"]
    }
  },
  AccordionSection: {
    path: "./runtime/components/bootstrap/accordion/AccordionSection.vue",
    category: "bootstrap",
    spec: {
      parents: ["Accordion"]
    }
  },
  Alert: {
    path: "./runtime/components/bootstrap/alert/Alert",
    category: "bootstrap",
    examples: {
      test: {
        text: "Alert"
      }
    }
  },
  AlertHeading: {
    path: "./runtime/components/bootstrap/alert/AlertHeading",
    category: "bootstrap",
    spec: {
      parents: ["Alert"]
    }
  },
  AlertLink: {
    path: "./runtime/components/bootstrap/alert/AlertLink.vue",
    category: "bootstrap",
    spec: {
      parents: ["Alert"]
    }
  },
  Backdrop: {
    path: "./runtime/components/bootstrap/backdrop/Backdrop",
    category: "bootstrap",
    spec: {
      parents: ["Offcanvas", "Modal"]
    }
  },
  Badge: {
    path: "./runtime/components/bootstrap/badge/Badge",
    category: "bootstrap",
    examples: {
      test: {
        text: "Badge"
      }
    }
  },
  Breadcrumb: {
    path: "./runtime/components/bootstrap/breadcrumb/Breadcrumb",
    category: "bootstrap",
    examples: {
      test: {
        items: [
          {
            component: "BreadcrumbItem",
            text: "BreadcrumbItem 1"
          },
          {
            component: "BreadcrumbItem",
            text: "BreadcrumbItem 2"
          }
        ]
      }
    }
  },
  BreadcrumbItem: {
    path: "./runtime/components/bootstrap/breadcrumb/BreadcrumbItem.vue",
    category: "bootstrap",
    spec: {
      parents: ["Breadcrumb"]
    }
  },
  ButtonGroup: {
    path: "./runtime/components/bootstrap/button/ButtonGroup",
    category: "bootstrap"
  },
  ButtonToolbar: {
    path: "./runtime/components/bootstrap/button/ButtonToolbar",
    category: "bootstrap"
  },
  CheckboxToggleButton: {
    path: "./runtime/components/bootstrap/button/CheckboxToggleButton",
    category: "bootstrap"
  },
  CloseButton: {
    path: "./runtime/components/bootstrap/button/CloseButton",
    category: "bootstrap"
  },
  RadioToggleButton: {
    path: "./runtime/components/bootstrap/button/RadioToggleButton",
    category: "bootstrap"
  },
  Card: {
    path: "./runtime/components/bootstrap/card/Card",
    category: "bootstrap"
  },
  CardBody: {
    path: "./runtime/components/bootstrap/card/CardBody",
    category: "bootstrap",
    spec: {
      parents: ["Card"]
    }
  },
  CardFooter: {
    path: "./runtime/components/bootstrap/card/CardFooter",
    category: "bootstrap",
    spec: {
      parents: ["Card"]
    }
  },
  CardGroup: {
    path: "./runtime/components/bootstrap/card/CardGroup",
    category: "bootstrap"
  },
  CardHeader: {
    path: "./runtime/components/bootstrap/card/CardHeader",
    category: "bootstrap",
    spec: {
      parents: ["Card"]
    }
  },
  CardImg: {
    path: "./runtime/components/bootstrap/card/CardImg",
    category: "bootstrap",
    spec: {
      parents: ["Card"]
    }
  },
  CardImgBottom: {
    path: "./runtime/components/bootstrap/card/CardImgBottom",
    category: "bootstrap",
    spec: {
      parents: ["Card"]
    }
  },
  CardImgOverlay: {
    path: "./runtime/components/bootstrap/card/CardImgOverlay",
    category: "bootstrap",
    spec: {
      parents: ["Card"]
    }
  },
  CardImgTop: {
    path: "./runtime/components/bootstrap/card/CardImgTop",
    category: "bootstrap",
    spec: {
      parents: ["Card"]
    }
  },
  CardLink: {
    path: "./runtime/components/bootstrap/card/CardLink.vue",
    category: "bootstrap",
    spec: {
      parents: ["CardBody", "CardFooter", "CardHeader"]
    }
  },
  CardSubTitle: {
    path: "./runtime/components/bootstrap/card/CardSubTitle",
    category: "bootstrap",
    spec: {
      parents: ["CardBody", "CardFooter", "CardHeader"]
    }
  },
  CardText: {
    path: "./runtime/components/bootstrap/card/CardText",
    category: "bootstrap",
    spec: {
      parents: ["CardBody", "CardFooter", "CardHeader"]
    }
  },
  CardTitle: {
    path: "./runtime/components/bootstrap/card/CardTitle",
    category: "bootstrap",
    spec: {
      parents: ["CardBody", "CardFooter", "CardHeader"]
    }
  },
  Carousel: {
    path: "./runtime/components/bootstrap/carousel/Carousel",
    category: "bootstrap"
  },
  CarouselCaption: {
    path: "./runtime/components/bootstrap/carousel/CarouselCaption",
    category: "bootstrap",
    spec: {
      parents: ["CarouselItem"]
    }
  },
  CarouselInner: {
    path: "./runtime/components/bootstrap/carousel/CarouselInner",
    category: "bootstrap",
    spec: {
      parents: ["Carousel"]
    }
  },
  CarouselItem: {
    path: "./runtime/components/bootstrap/carousel/CarouselItem",
    category: "bootstrap",
    spec: {
      parents: ["CarouselInner"]
    }
  },
  CarouselItemImage: {
    path: "./runtime/components/bootstrap/carousel/CarouselItemImage",
    category: "bootstrap",
    spec: {
      parents: ["CarouselItem"]
    }
  },
  Collapse: {
    path: "./runtime/components/bootstrap/collapse/Collapse",
    category: "bootstrap"
  },
  CollapseButton: {
    path: "./runtime/components/bootstrap/collapse/CollapseButton",
    category: "bootstrap"
  },
  Flex: {
    path: "./runtime/components/bootstrap/cssgrid/Flex",
    category: "bootstrap"
  },
  FlexItem: {
    path: "./runtime/components/bootstrap/cssgrid/FlexItem",
    category: "bootstrap"
  },
  Grid: {
    path: "./runtime/components/bootstrap/cssgrid/Grid",
    category: "bootstrap"
  },
  GridCol: {
    path: "./runtime/components/bootstrap/cssgrid/GridCol",
    category: "bootstrap"
  },
  Dropdown: {
    path: "./runtime/components/bootstrap/dropdown/Dropdown",
    category: "bootstrap"
  },
  DropdownItem: {
    path: "./runtime/components/bootstrap/dropdown/DropdownItem.vue",
    category: "bootstrap"
  },
  DropdownItemDivider: {
    path: "./runtime/components/bootstrap/dropdown/DropdownItemDivider",
    category: "bootstrap"
  },
  DropdownItemText: {
    path: "./runtime/components/bootstrap/dropdown/DropdownItemText",
    category: "bootstrap"
  },
  DropdownMenu: {
    path: "./runtime/components/bootstrap/dropdown/DropdownMenu",
    category: "bootstrap"
  },
  DropdownToggle: {
    path: "./runtime/components/bootstrap/dropdown/DropdownToggle",
    category: "bootstrap"
  },
  DropdownToggleSplit: {
    path: "./runtime/components/bootstrap/dropdown/DropdownToggleSplit",
    category: "bootstrap"
  },
  Col: {
    path: "./runtime/components/bootstrap/grid/Col",
    category: "bootstrap"
  },
  Container: {
    path: "./runtime/components/bootstrap/grid/Container",
    category: "bootstrap"
  },
  Row: {
    path: "./runtime/components/bootstrap/grid/Row",
    category: "bootstrap"
  },
  ListGroup: {
    path: "./runtime/components/bootstrap/list-group/ListGroup",
    category: "bootstrap"
  },
  ListGroupItem: {
    path: "./runtime/components/bootstrap/list-group/ListGroupItem",
    category: "bootstrap"
  },
  ListGroupItemAction: {
    path: "./runtime/components/bootstrap/list-group/ListGroupItemAction.vue",
    category: "bootstrap"
  },
  ListGroupItemLabel: {
    path: "./runtime/components/bootstrap/list-group/ListGroupItemLabel",
    category: "bootstrap"
  },
  ListGroupList: {
    path: "./runtime/components/bootstrap/list-group/ListGroupList",
    category: "bootstrap"
  },
  Modal: {
    path: "./runtime/components/bootstrap/modal/Modal",
    category: "bootstrap"
  },
  ModalBody: {
    path: "./runtime/components/bootstrap/modal/ModalBody",
    category: "bootstrap"
  },
  ModalContent: {
    path: "./runtime/components/bootstrap/modal/ModalContent",
    category: "bootstrap"
  },
  ModalDialog: {
    path: "./runtime/components/bootstrap/modal/ModalDialog",
    category: "bootstrap"
  },
  ModalFooter: {
    path: "./runtime/components/bootstrap/modal/ModalFooter",
    category: "bootstrap"
  },
  ModalHeader: {
    path: "./runtime/components/bootstrap/modal/ModalHeader",
    category: "bootstrap"
  },
  ModalTitle: {
    path: "./runtime/components/bootstrap/modal/ModalTitle",
    category: "bootstrap"
  },
  ModalToggleButton: {
    path: "./runtime/components/bootstrap/modal/ModalToggleButton",
    category: "bootstrap"
  },
  Nav: {
    path: "./runtime/components/bootstrap/nav/Nav",
    category: "bootstrap"
  },
  NavItem: {
    path: "./runtime/components/bootstrap/nav/NavItem",
    category: "bootstrap"
  },
  NavItemDropdown: {
    path: "./runtime/components/bootstrap/nav/NavItemDropdown",
    category: "bootstrap"
  },
  NavItemDropdownToggle: {
    path: "./runtime/components/bootstrap/nav/NavItemDropdownToggle",
    category: "bootstrap"
  },
  NavLink: {
    path: "./runtime/components/bootstrap/nav/NavLink.vue",
    category: "bootstrap"
  },
  NavList: {
    path: "./runtime/components/bootstrap/nav/NavList",
    category: "bootstrap"
  },
  TabContent: {
    path: "./runtime/components/bootstrap/nav/TabContent",
    category: "bootstrap"
  },
  TabPane: {
    path: "./runtime/components/bootstrap/nav/TabPane",
    category: "bootstrap"
  },
  Navbar: {
    path: "./runtime/components/bootstrap/navbar/Navbar",
    category: "bootstrap"
  },
  NavbarBrand: {
    path: "./runtime/components/bootstrap/navbar/NavbarBrand",
    category: "bootstrap"
  },
  NavbarCollapse: {
    path: "./runtime/components/bootstrap/navbar/NavbarCollapse",
    category: "bootstrap"
  },
  NavbarNav: {
    path: "./runtime/components/bootstrap/navbar/NavbarNav",
    category: "bootstrap"
  },
  NavbarNavList: {
    path: "./runtime/components/bootstrap/navbar/NavbarNavList",
    category: "bootstrap"
  },
  NavbarText: {
    path: "./runtime/components/bootstrap/navbar/NavbarText",
    category: "bootstrap"
  },
  NavbarToggler: {
    path: "./runtime/components/bootstrap/navbar/NavbarToggler",
    category: "bootstrap"
  },
  Offcanvas: {
    path: "./runtime/components/bootstrap/offcanvas/Offcanvas",
    category: "bootstrap"
  },
  OffcanvasBody: {
    path: "./runtime/components/bootstrap/offcanvas/OffcanvasBody",
    category: "bootstrap"
  },
  OffcanvasHeader: {
    path: "./runtime/components/bootstrap/offcanvas/OffcanvasHeader",
    category: "bootstrap"
  },
  OffcanvasTitle: {
    path: "./runtime/components/bootstrap/offcanvas/OffcanvasTitle",
    category: "bootstrap"
  },
  PageItem: {
    path: "./runtime/components/bootstrap/pagination/PageItem",
    category: "bootstrap"
  },
  PageLink: {
    path: "./runtime/components/bootstrap/pagination/PageLink",
    category: "bootstrap"
  },
  Pagination: {
    path: "./runtime/components/bootstrap/pagination/Pagination",
    category: "bootstrap"
  },
  Progress: {
    path: "./runtime/components/bootstrap/progress/Progress",
    category: "bootstrap"
  },
  ProgressBar: {
    path: "./runtime/components/bootstrap/progress/ProgressBar",
    category: "bootstrap"
  },
  Spinner: {
    path: "./runtime/components/bootstrap/spinner/Spinner",
    category: "bootstrap"
  },
  Toast: {
    path: "./runtime/components/bootstrap/toast/Toast",
    category: "bootstrap"
  },
  ToastBody: {
    path: "./runtime/components/bootstrap/toast/ToastBody",
    category: "bootstrap"
  },
  ToastContainer: {
    path: "./runtime/components/bootstrap/toast/ToastContainer",
    category: "bootstrap"
  },
  ToastHeader: {
    path: "./runtime/components/bootstrap/toast/ToastHeader",
    category: "bootstrap"
  },
  Vr: {
    path: "./runtime/components/bootstrap/vertical-rule/VerticalRule",
    category: "bootstrap"
  },
  Avatar: {
    path: "./runtime/components/extend/avatar/Avatar",
    category: "extend",
    props: {
      circle: {
        type: Boolean,
        default: false
      },
      size: {
        type: String,
        default: ""
      },
      imgSrc: {
        type: String,
        default: ""
      },
      imgAlt: {
        type: String,
        default: ""
      },
      initials: {
        type: String,
        default: ""
      },
      color: {
        type: String,
        default: ""
      },
      status: {
        type: [Boolean, String],
        default: false
      },
      statusColor: {
        type: String,
        default: ""
      },
      icon: {
        type: String,
        default: ""
      },
      divider: {
        type: [Boolean, String],
        default: false
      },
      objectFit: {
        type: [String]
      }
    }
  },
  BackToTop: {
    path: "./runtime/components/extend/back-to-top/BackToTop.vue",
    category: "extend"
  },
  Background: {
    path: "./runtime/components/extend/background/Background.vue",
    category: "extend"
  },
  Calendar: {
    path: "./runtime/components/extend/calendar/Calendar.vue",
    category: "extend"
  },
  DatePicker: {
    path: "./runtime/components/extend/calendar/DatePicker.vue",
    category: "extend"
  },
  CodeHighlighter: {
    path: "./runtime/components/extend/code/CodeHighlighter.vue",
    category: "extend"
  },
  EventViewer: {
    path: "./runtime/components/extend/code/EventViewer.vue",
    category: "extend"
  },
  ColorModeNavItemDropdown: {
    path: "./runtime/components/extend/color-mode/ColorModeNavItemDropdown.vue",
    category: "extend"
  },
  // "Divider": {
  //  path: "./runtime/components/extend/divider/Divider.vue",
  //  category: "extend",
  // },
  // "HTVDivider": {
  //  path: "./runtime/components/extend/divider/HTVDivider.vue",
  //  category: "extend",
  // },
  ResponsiveRule: {
    path: "./runtime/components/extend/divider/ResponsiveRule.vue",
    category: "extend"
  },
  LocaleNavItemDropdown: {
    path: "./runtime/components/extend/dynamic-route/LocaleNavItemDropdown.vue",
    category: "extend"
  },
  Localization: {
    path: "./runtime/components/extend/dynamic-route/Localization.vue",
    category: "extend"
  },
  BarChart: {
    path: "./runtime/components/extend/echart/BarChart.vue",
    category: "extend"
  },
  LineChart: {
    path: "./runtime/components/extend/echart/LineChart.vue",
    category: "extend"
  },
  PieChart: {
    path: "./runtime/components/extend/echart/PieChart.vue",
    category: "extend"
  },
  Fixed: {
    path: "./runtime/components/extend/fixed/Fixed.vue",
    category: "extend"
  },
  GridArea: {
    path: "./runtime/components/extend/grid-template/GridArea",
    category: "extend",
    spec: {
      parents: ["GridTemplate"]
    }
  },
  GridTemplate: {
    path: "./runtime/components/extend/grid-template/GridTemplate.vue",
    category: "extend",
    spec: {
      layout: true
    }
  },
  HeadingBlock: {
    path: "./runtime/components/extend/heading/HeadingBlock.vue",
    category: "extend"
  },
  HeroRow: {
    path: "./runtime/components/extend/hero-row/HeroRow.vue",
    category: "extend"
  },
  IconBox: {
    path: "./runtime/components/extend/icon-box/IconBox",
    category: "extend"
  },
  Intro: {
    path: "./runtime/components/extend/intro/Intro.vue",
    category: "extend"
  },
  JsonView: {
    path: "./runtime/components/extend/json-view/JsonView.vue",
    category: "extend"
  },
  PageLayout: {
    path: "./runtime/components/extend/layout/PageLayout.vue",
    category: "extend"
  },
  FloatingNavbar: {
    path: "./runtime/components/extend/navbar/FloatingNavbar.vue",
    category: "extend"
  },
  SlideNavbar: {
    path: "./runtime/components/extend/navbar/SlideNavbar.vue",
    category: "extend"
  },
  StickyToggleNavbar: {
    path: "./runtime/components/extend/navbar/StickyToggleNavbar.vue",
    category: "extend"
  },
  PasswordStrength: {
    path: "./runtime/components/extend/password-strength/PasswordStrength.vue",
    category: "extend"
  },
  Paper: {
    path: "./runtime/components/extend/paper/Paper.vue",
    category: "extend"
  },
  Rating: {
    path: "./runtime/components/extend/rating/Rating.vue",
    category: "extend"
  },
  Sidebar: {
    path: "./runtime/components/extend/sidebar/Sidebar.vue",
    category: "extend"
  },
  Sortable: {
    path: "./runtime/components/extend/sortable/Sortable.vue",
    category: "extend",
    spec: {
      parents: ["ListGroupList"]
    }
  },
  ComponentsCatalog: {
    path: "./runtime/components/spec/ComponentsCatalog.vue",
    category: "spec"
  },
  UseBootstrap: {
    path: "./runtime/components/spec/UseBootstrap.vue",
    category: "spec"
  },
  SpecBinding: {
    path: "./runtime/components/spec/SpecBinding.vue",
    category: "spec"
  },
  StackGroup: {
    path: "./runtime/components/extend/stack-group/StackGroup.vue",
    category: "extend"
  },
  StackGroupItem: {
    path: "./runtime/components/extend/stack-group/StackGroupItem.vue",
    category: "extend"
  },
  StatusIndicator: {
    path: "./runtime/components/extend/status-indicator/StatusIndicator",
    category: "extend"
  },
  Step: {
    path: "./runtime/components/extend/step/Step.vue",
    category: "extend"
  },
  StepItem: {
    path: "./runtime/components/extend/step/StepItem.vue",
    category: "extend"
  },
  Swiper: {
    path: "./runtime/components/extend/swiper/Swiper.vue",
    category: "extend"
  },
  TiptapEditor: {
    path: "./runtime/components/extend/tiptap/TiptapEditor.vue",
    category: "extend"
  },
  CollapseToc: {
    path: "./runtime/components/extend/toc/CollapseToc.vue",
    category: "extend"
  },
  Toc: {
    path: "./runtime/components/extend/toc/Toc.vue",
    category: "extend"
  },
  Typed: {
    path: "./runtime/components/extend/typed/Typed",
    category: "extend",
    examples: {
      test: {
        attrs: {
          strings: ["Hello", "World"]
        }
      }
    }
  },
  Blockquote: {
    path: "./runtime/components/html-block/blockquote/Blockquote",
    category: "html"
  },
  BlockquoteFigure: {
    path: "./runtime/components/html-block/blockquote/BlockquoteFigure",
    category: "html"
  },
  BlockquoteFooter: {
    path: "./runtime/components/html-block/blockquote/BlockquoteFooter",
    category: "html"
  },
  Figure: {
    path: "./runtime/components/html-block/figure/Figure",
    category: "html"
  },
  FigureCaption: {
    path: "./runtime/components/html-block/figure/FigureCaption",
    category: "html"
  },
  FigureImage: {
    path: "./runtime/components/html-block/figure/FigureImage",
    category: "html"
  },
  ColFormLabel: {
    path: "./runtime/components/html-block/form/ColFormLabel",
    category: "html"
  },
  ColFormLegend: {
    path: "./runtime/components/html-block/form/ColFormLegend",
    category: "html"
  },
  Datalist: {
    path: "./runtime/components/html-block/form/Datalist",
    category: "html"
  },
  Fieldset: {
    path: "./runtime/components/html-block/form/Fieldset",
    category: "html"
  },
  Form: {
    path: "./runtime/components/html-block/form/Form",
    category: "html"
  },
  FormCheck: {
    path: "./runtime/components/html-block/form/FormCheck",
    category: "html"
  },
  FormCheckInput: {
    path: "./runtime/components/html-block/form/FormCheckInput.vue",
    category: "html"
  },
  FormCheckLabel: {
    path: "./runtime/components/html-block/form/FormCheckLabel",
    category: "html"
  },
  FormColor: {
    path: "./runtime/components/html-block/form/FormColor.vue",
    category: "html"
  },
  FormControl: {
    path: "./runtime/components/html-block/form/FormControl",
    category: "html"
  },
  FormFile: {
    path: "./runtime/components/html-block/form/FormFile.vue",
    category: "html"
  },
  FormFloating: {
    path: "./runtime/components/html-block/form/FormFloating",
    category: "html"
  },
  FormInput: {
    path: "./runtime/components/html-block/form/FormInput.vue",
    category: "html"
  },
  FormLabel: {
    path: "./runtime/components/html-block/form/FormLabel",
    category: "html"
  },
  FormRange: {
    path: "./runtime/components/html-block/form/FormRange.vue",
    category: "html"
  },
  FormSelect: {
    path: "./runtime/components/html-block/form/FormSelect.vue",
    category: "html"
  },
  FormSwitch: {
    path: "./runtime/components/html-block/form/FormSwitch",
    category: "html"
  },
  FormText: {
    path: "./runtime/components/html-block/form/FormText",
    category: "html"
  },
  FormTextarea: {
    path: "./runtime/components/html-block/form/FormTextarea.vue",
    category: "html"
  },
  InputGroup: {
    path: "./runtime/components/html-block/form/InputGroup",
    category: "html"
  },
  InputGroupText: {
    path: "./runtime/components/html-block/form/InputGroupText",
    category: "html"
  },
  Label: {
    path: "./runtime/components/html-block/form/Label",
    category: "html"
  },
  Legend: {
    path: "./runtime/components/html-block/form/Legend",
    category: "html"
  },
  Option: {
    path: "./runtime/components/html-block/form/Option",
    category: "html"
  },
  ValidFeedback: {
    path: "./runtime/components/html-block/form/ValidFeedback.vue",
    category: "html"
  },
  ValidTooltip: {
    path: "./runtime/components/html-block/form/ValidTooltip.vue",
    category: "html"
  },
  Block: {
    path: "./runtime/components/html-block/Block",
    category: "html"
  },
  Dd: {
    path: "./runtime/components/html-block/DefinitionDescription",
    category: "html"
  },
  Dl: {
    path: "./runtime/components/html-block/DefinitionList",
    category: "html"
  },
  Dt: {
    path: "./runtime/components/html-block/DefinitionTerm",
    category: "html"
  },
  Div: {
    path: "./runtime/components/html-block/Div",
    category: "html"
  },
  Footer: {
    path: "./runtime/components/html-block/Footer",
    category: "html"
  },
  Hr: {
    path: "./runtime/components/html-block/Hr",
    category: "html"
  },
  Img: {
    path: "./runtime/components/html-block/Image",
    category: "html"
  },
  Image: {
    alias: "Img",
    category: "extend"
  },
  Input: {
    path: "./runtime/components/html-block/HtmlInput.vue",
    category: "html"
  },
  Ul: {
    path: "./runtime/components/html-block/List",
    category: "html"
  },
  List: {
    alias: "Ul",
    category: "extend"
  },
  Li: {
    path: "./runtime/components/html-block/ListItem.vue",
    category: "html"
  },
  ListItem: {
    alias: "Li",
    category: "extend"
  },
  Main: {
    path: "./runtime/components/html-block/Main",
    category: "html"
  },
  Ol: {
    path: "./runtime/components/html-block/OrderedList",
    category: "html"
  },
  P: {
    path: "./runtime/components/html-block/Paragraph",
    category: "html"
  },
  Paragraph: {
    alias: "P",
    category: "extend"
  },
  Picture: {
    path: "./runtime/components/html-block/Picture",
    category: "html"
  },
  Select: {
    path: "./runtime/components/html-block/Select.vue",
    category: "html"
  },
  Textarea: {
    path: "./runtime/components/html-block/Textarea.vue",
    category: "html"
  },
  A: {
    path: "./runtime/components/html-inline/Anchor",
    category: "html"
  },
  Anchor: {
    alias: "A",
    category: "extend"
  },
  Button: {
    path: "./runtime/components/html-inline/Button",
    category: "html"
  },
  H1: {
    path: "./runtime/components/html-inline/H1",
    category: "html"
  },
  H2: {
    path: "./runtime/components/html-inline/H2",
    category: "html"
  },
  H3: {
    path: "./runtime/components/html-inline/H3",
    category: "html"
  },
  H4: {
    path: "./runtime/components/html-inline/H4",
    category: "html"
  },
  H5: {
    path: "./runtime/components/html-inline/H5",
    category: "html"
  },
  H6: {
    path: "./runtime/components/html-inline/H6",
    category: "html"
  },
  H: {
    path: "./runtime/components/html-inline/Headings",
    category: "html"
  },
  Headings: {
    alias: "H",
    category: "extend"
  },
  Span: {
    path: "./runtime/components/html-inline/Inline",
    category: "html"
  },
  Inline: {
    alias: "Span",
    category: "extend"
  },
  Small: {
    path: "./runtime/components/html-inline/Small",
    category: "html"
  },
  Strong: {
    path: "./runtime/components/html-inline/Strong",
    category: "html"
  },
  Caption: {
    path: "./runtime/components/html-table/Caption",
    category: "html"
  },
  Table: {
    path: "./runtime/components/html-table/Table",
    category: "html"
  },
  Tbody: {
    path: "./runtime/components/html-table/Tbody",
    category: "html"
  },
  Td: {
    path: "./runtime/components/html-table/Td",
    category: "html"
  },
  Tfoot: {
    path: "./runtime/components/html-table/Tfoot",
    category: "html"
  },
  Th: {
    path: "./runtime/components/html-table/Th",
    category: "html"
  },
  Thead: {
    path: "./runtime/components/html-table/Thead",
    category: "html"
  },
  Tr: {
    path: "./runtime/components/html-table/Tr",
    category: "html"
  },
  Icon: {
    path: "./runtime/components/icon/BIcon",
    category: "icon"
  },
  ActionState: {
    path: "./runtime/components/integration/ActionState.vue",
    category: "integration"
  },
  ArrayQuery: {
    path: "./runtime/components/integration/ArrayQuery.vue",
    category: "integration"
  },
  DataTable: {
    path: "./runtime/components/integration/DataTable.vue",
    category: "integration"
  },
  DataTableBody: {
    path: "./runtime/components/integration/DataTableBody.vue",
    category: "integration"
  },
  DataTableFooter: {
    path: "./runtime/components/integration/DataTableFooter.vue",
    category: "integration"
  },
  DataTableHeader: {
    path: "./runtime/components/integration/DataTableHeader.vue",
    category: "integration"
  },
  DataTableHeaderRowFilter: {
    path: "./runtime/components/integration/DataTableHeaderRowFilter.vue",
    category: "integration"
  },
  DataTableHeaderRowSorter: {
    path: "./runtime/components/integration/DataTableHeaderRowSorter.vue",
    category: "integration"
  },
  FullTextFilter: {
    path: "./runtime/components/integration/FullTextFilter.vue",
    category: "integration"
  },
  ViewState: {
    path: "./runtime/components/integration/ViewState.vue",
    category: "integration",
    examples: {
      test: {
        attrs: {
          src: "query://uid"
        }
      }
    }
  },
  SwiperSlide: {
    export: "SwiperSlide",
    path: "swiper/vue",
    category: "swiper"
  },
  UseWindowSize: {
    export: "UseWindowSize",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseWindowFocus: {
    export: "UseWindowFocus",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseVirtualList: {
    export: "UseVirtualList",
    path: "@vueuse/components",
    category: "vueuse"
  },
  UseTimestamp: {
    export: "UseTimestamp",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseTimeAgo: {
    export: "UseTimeAgo",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseScreenSafeArea: {
    export: "UseScreenSafeArea",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UsePreferredReducedMotion: {
    export: "UsePreferredReducedMotion",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UsePreferredLanguages: {
    export: "UsePreferredLanguages",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UsePreferredDark: {
    export: "UsePreferredDark",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UsePreferredContrast: {
    export: "UsePreferredContrast",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UsePreferredColorScheme: {
    export: "UsePreferredColorScheme",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UsePointerLock: {
    export: "UsePointerLock",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UsePointer: {
    export: "UsePointer",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UsePageLeave: {
    export: "UsePageLeave",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseOnline: {
    export: "UseOnline",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseOffsetPagination: {
    export: "UseOffsetPagination",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseObjectUrl: {
    export: "UseObjectUrl",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseNow: {
    export: "UseNow",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseNetwork: {
    export: "UseNetwork",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseMousePressed: {
    export: "UseMousePressed",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseMouseInElement: {
    export: "UseMouseInElement",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseIdle: {
    export: "UseIdle",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseGeolocation: {
    export: "UseGeolocation",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseFullscreen: {
    export: "UseFullscreen",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseEyeDropper: {
    export: "UseEyeDropper",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseElementVisibility: {
    export: "UseElementVisibility",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseElementSize: {
    export: "UseElementSize",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseElementBounding: {
    export: "UseElementBounding",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseDraggable: {
    export: "UseDraggable",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseDocumentVisibility: {
    export: "UseDocumentVisibility",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseDevicesList: {
    export: "UseDevicesList",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseDevicePixelRatio: {
    export: "UseDevicePixelRatio",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseDeviceOrientation: {
    export: "UseDeviceOrientation",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseDeviceMotion: {
    export: "UseDeviceMotion",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseClipboard: {
    export: "UseClipboard",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseBattery: {
    export: "UseBattery",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  UseActiveElement: {
    export: "UseActiveElement",
    path: "@vueuse/components",
    category: "vueuse"
    // mode: "client",
  },
  OnClickOutside: {
    export: "OnClickOutside",
    path: "@vueuse/components",
    category: "vueuse",
    mode: "client"
  }
};
